src/App.tsx

```diff
@@ -1,27 +1,67 @@
-import React from 'react';
+import React, { useMemo, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  const handleLoadingAllGoods = () => {
+    getAll()
+      .then(setGoods)
+      .catch(error => setErrorMessage(error.message));
+  };
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const handleLoading5First = () => {
+    get5First()
+      .then(setGoods)
+      .catch(error => setErrorMessage(error.message));
+  };
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  const handleLoadingRed = () => {
+    getRedGoods()
+      .then(setGoods)
+      .catch(error => setErrorMessage(error.message));
+  };
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  const visibleGoods = useMemo(() => {
+    return goods;
+  }, [goods]);
```

Comment:
using `useMemo` with the same input and output values does not make sense as it will not provide any optimization or benefit in this scenario